### ISSUE TITLE
Simple way to open new directory

### ISSUE THREAD

**Issue by jplindstrom** [Alexander-Miller/treemacs#681]
(from the Reddit thread at https://www.reddit.com/r/emacs/comments/gponrz/treemacs_news_v2/)

Previously when I wanted to open treemacs in a new directory I could just close the old Treemacs buffer and then run M-x treemacs in e.g. another dired buffer to start there. Now that doesn't work, it opens up the previous place.

One thing I did try was to see if `C-u M-x treemacs` would do this. I would expect that to let me enter a path to open, with a default of the cwd of the current buffer.

What's the simplest / idiomatic way of just starting in a new place in the file system? It's not clear after having read the docs.

**Comment by Alexander-Miller** [comment]
There's commands that do this for projects,I will need to implement new commands do it for the current directory:
 - `treemacs-add-and-display-current-directory`
 - `treemacs-display-current-directory-exclusively`

> One thing I did try was to see if C-u M-x treemacs would do this. I would expect that to let me enter a path to open, with a default of the cwd of the current buffer.

By default treemacs operates on the assumption of you having multiple projects, so entering a new path with a prefix arg does not really fit (as of now the prefix arg has no effect). For now the closest features to this are
 - `treemacs-add-and-display-current-project`
 - `treemacs-display-current-project-exclusively`

or using `h` & `l` when there's only a single project.

**Comment by exp2ipi** [comment]
Another vote for this feature.  I am a pretty light treemacs user and while I understand it's powerful and feature-ful it would be great if it did something dumb but understandable by default -- example: `M-x treemacs` could show the root of the current project (if in a buffer that's part of a projectile project), or the directory of the current buffer. As it stands, it defaulted to my `.emacs.d` as a root directory, and even after browsing the docs I am not entirely sure what the whole workspace/project concept is about.

tl;dr: be kind to users that just want to have a directory tree shown and learn the bells and whistles gradually :)

**Comment by Alexander-Miller** [comment]
For the `treemacs` entry point I have been thinking about going the way of clippy when it's called for the first time - open up a buffer with the same org-mode setup used for `treemacs-edit-workspaces`, but also a clippy dialog like the one you see when you manually call `treemacs-mode` that explain's what's what.

Do you think that would help?

**Comment by exp2ipi** [comment]
FWIW it would be nice to keep things as simple and friction-free for the end user; in that vein I think OP's idea of using C-u prefix to open treemacs in a directory is a good one. I'm not familiar with clippy but some cheat-sheet popup upon first invocation is reasonable. Asking the user to configure things right off the bat is IMO less than ideal however.

**Comment by Alexander-Miller** [comment]
Stayin alive.

**Comment by Alexander-Miller** [comment]
Stayin alive.

**Comment by Alexander-Miller** [comment]
I've now added the `treemacs-select-directory` command that lets you pick a single directory to show in treemacs.

**Comment by nicolasshu** [comment]
Oh thank you so much! This is great!

**Comment by Natzgun** [comment]
I am currently trying to use Treemacs, it is simply wonderful, regarding this problem I made a configuration so that when I change directories it shows me exclusively the directory I have changed to, the only problem is that it leaves the file tree open, maybe There is a better way to do that, I haven't been able to achieve it yet.
```
defun my/treemacs-display-current-project-exclusively ()
  (treemacs-display-current-project-exclusively))
 (add-hook 'find-file-hook #'my/treemacs-display-current-project-exclusively)

```

**Comment by Alexander-Miller** [comment]
> the only problem is that it leaves the file tree open

Try to `(setf treemacs-expand-after-init nil)`.

**Comment by Natzgun** [comment]
A big mistake of mine, while I was reviewing other Issues I realized that someone had the same problem as me and there was already the implementation.
I solved it with:
```
treemacs-project-follow-mode
```
Now I'll just configure the delay, thank you very much :)

**Comment by Natzgun** [comment]
I've tried using:
```
treemacs-project-follow-mode
```
But it doesn't work correctly when I open treemacs it still shows me the files from the previous project, when I do spc h r r, there it just changes to the current project, but when I go to another project it still shows me the previous one, and again I have to do spc h r r, dunno what am I doing wrong.

**Comment by Alexander-Miller** [comment]
First I need to know what that keybind does. It looks like something from spacemacs, which I don't use anymore, so I've nothing to go by here.

Second how do you enable that mode? I've seen plenty of people just change the indicator variable instead of properly starting a minor mode, so make sure you call the function and don't just set `treemacs-project-follow-mode`.
